agent_build_refactored/scripts/get_github_actions_job_matrices.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2014-2022 Scalyr Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+This script generates job matrices for GitHub Actions. This is needed because we do a "full" run on GHA
+when we push on master or create a pull request, in other cases (e.g. just custom development branch)
+we just run limited set of tests to reduce number of jobs for active development phase.
+"""
+
+import json
+import os
+import sys
+import pathlib as pl
+
+# This file can be executed as script. Add source root to the PYTHONPATH in order to be able to import
+# local packages. All such imports also have to be done after that.
+sys.path.append(str(pl.Path(__file__).parent.parent.parent))
+
+from agent_build_refactored.tools.constants import SOURCE_ROOT
+from agent_build_refactored.tools.runner import Runner
+from agent_build_refactored.docker_image_builders import IMAGES_PYTHON_VERSION
+
+from agent_build_refactored.docker_image_builders import (
+    DEBIAN_IMAGE_BUILDERS,
+    ALL_IMAGE_BUILDERS,
+)
+
+# We expect some info from the GitHub actions context to determine if the run is limited or not.
+GITHUB_EVENT_NAME = os.environ.get("GITHUB_EVENT_NAME", "")
+GITHUB_BASE_REF = os.environ.get("GITHUB_BASE_REF", "")
+GITHUB_REF_TYPE = os.environ.get("GITHUB_REF_TYPE", "")
+GITHUB_REF_NAME = os.environ.get("GITHUB_REF_NAME", "")
+
+limited_run = True
+
+# We do a full workflow run on:
+# pull request against the 'master' branch.
+if GITHUB_EVENT_NAME == "pull_request" and GITHUB_BASE_REF == "master":
+    limited_run = False
+# push to the 'master' branch
+elif (
+    GITHUB_EVENT_NAME == "push"
+    and GITHUB_REF_TYPE == "branch"
+    and GITHUB_REF_NAME == "master"
+):
+    limited_run = False
+
+# push to a "production" tag.
+elif GITHUB_EVENT_NAME == "push" and GITHUB_REF_TYPE == "tag":
+    version_file_path = SOURCE_ROOT / "VERSION"
+    current_version = version_file_path.read_text().strip()
+    if GITHUB_REF_NAME == f"v{current_version}":
+        limited_run = False
+
+# Set collection of builders that will be used during the CI/Cd workflow run.
+if limited_run:
+    used_image_builders = DEBIAN_IMAGE_BUILDERS[:]
```

Comment:
So this means we will only run Alpine builders on master run? Does this means that we won't notice if PR breaks Alpine image until it's merged into master?

Reply:
We should notice that on PR creation, since the it also has to initiate the "full" run.